Settle on convention of version name for snapshot builds

KAction opened the pull request:
There are currently two conventions for naming derivations that build not from
upstream release, but from arbitrary git commit -- either

{ pname = "foo-unstable"; version = "1970-01-01"; }

or

{ pname = "foo"; version = "unstable-1970-01-01"; }

Both of them result in same output path name, but second version is blessed by
nixpkgs manual. This commit mechanically converts packages that use first
convention to use second convention in cases when ${version} is not referenced
anywhere else in default.nix

=> https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

FRidh commented:
There is a RFC on this topic https://github.com/NixOS/rfcs/pull/107. It is my opinion both can be valid depending on whether a stable attribute exists or not.

KAction commented:
I love this RFC (mostly, initial release imho should be `0~foo`, not `0+foo`), but until then it would be nice to adhere to current standard, I think.

Artturin reviewed (approved):
LGTM other than that the commit msg should be `treewide: move unstable suffix from pname to version`

KAction commented:
@Artturin Updated commit message.

Commit:
treewide: move unstable suffix from pname to version

There are currently two conventions for naming derivations that build not from
upstream release, but from arbitrary git commit -- either

{ pname = "foo-unstable"; version = "1970-01-01"; }

or

{ pname = "foo"; version = "unstable-1970-01-01"; }

Both of them result in same output path name, but second version is blessed by
nixpkgs manual. This commit mechanically converts packages that use first
convention to use second convention in cases when ${version} is not referenced
anywhere else in default.nix

=> https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming

jopejoe1 commented:
We now have a versioning schema for non release builds.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming
